docs/standard/io/pipelines.md

```diff
@@ -334,6 +336,18 @@ The previous method of writing uses the buffers provided by the `PipeWriter`. It
 * Calling `GetMemory` or `GetSpan` while there's an incomplete call to `FlushAsync` isn't safe.
 * Calling `Complete` or `CompleteAsync` while there's unflushed data can result in memory corruption.
 
+## Tips for using PipeReader and PipeWriter
+
+The following tips will help you use the <xref:System.IO.Pipelines> classes successfully:
+
+* Always complete the [PipeReader](xref:System.IO.Pipelines.PipeReader.Complete%2A?displayProperty=nameWithType) and [PipeWriter](xref:System.IO.Pipelines.PipeWriter.Complete%2A?displayProperty=nameWithType), including an exception where applicable.
+* Always call <xref:System.IO.Pipelines.PipeReader.AdvanceTo%2A?displayProperty=nameWithType> after calling <xref:System.IO.Pipelines.PipeReader.ReadAsync%2A?displayProperty=nameWithType>.
+* Periodically `await` <xref:System.IO.Pipelines.PipeWriter.FlushAsync%2A?displayProperty=nameWithType> while writing, and always check <xref:System.IO.Pipelines.FlushResult.IsCompleted?displayProperty=nameWithType>. Abort writing if `IsCompleted` is `true`, as that indicates the reader is completed and no longer cares about what is written.
```

Comment:
Do we have any guidance on how often this should be done?

Reply:
This is all we had in the issue - do you recall who/where the guidance came from?